components/locid_transform/src/fallback/mod.rs

```diff
@@ -350,32 +355,20 @@ impl LocaleFallbacker {
         }
     }
 
-    /// Creates an iterator based on a [`Locale`].
-    ///
-    /// When first initialized, the locale is normalized according to the fallback algorithm.
-    ///
-    /// For performance reasons, internally an [`icu_provider::DataLocale`] is used.
-    ///
-    /// [`Locale`]: icu_locid::Locale
-    pub fn fallback_for(
-        &self,
-        config: LocaleFallbackConfig,
-        locale: impl Into<DataLocale>,
-    ) -> LocaleFallbackIterator {
-        self.for_config(config).fallback_for(locale.into())
-    }
-
-    #[doc(hidden)]
+    /// Associates a configuration with this fallbacker.
+    #[inline]
     pub fn for_config(&self, config: LocaleFallbackConfig) -> LocaleFallbackerWithConfig {
         self.as_borrowed().for_config(config)
     }
 
-    #[doc(hidden)]
+    /// Derives a configuration from a [`DataKey`] and associates it
+    /// with this fallbacker.
+    #[inline]
```

Comment:
I had this as `doc(hidden)` because a `DataKey` is a data pipeline implementation detail, and not something that should necessarily be on `locid_transform`'s API. Clients can do `key.into()` themselves.

Reply:
> in most use cases, you start with a DataKey

Disagree. In most use cases *we* start with a `DataKey`, but users will use fallback for other purposes than building a fallback provider.

Reply:
Also `.into()` is not `const`

Reply:
ok done